app/components/elements/header_component.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Elements
+  # Applies an h# component with the expected styling.
+  class HeaderComponent < ApplicationComponent
+    def initialize(variant, value:, classes: [])
```

Comment:
I'd recommend calling the initial arg "level" and making it a kwarg for consistency: `def initialize(level:, value:, classes: [])`